pyo3-ffi/src/cpython/abstract_.rs

```diff
@@ -6,7 +6,7 @@ use crate::Py_buffer;
 
 #[cfg(Py_3_8)]
 use crate::pyport::PY_SSIZE_T_MAX;
-#[cfg(all(Py_3_8, not(PyPy)))]
+#[cfg(all(Py_3_8, not(any(PyPy, GraalPy))))]
```

Comment:
I think this will only really help if we have all three of them and our build system ensures that they are consistent, so that I can always reach for the one which fits my needs best.

Reply:
Agreed, I was thinking we would always emit exactly one of `CPython`, `PyPy`, or `GraalPy`

Reply:
I started playing with this but TBH given that `CPython` is essentially the reference implementation and the default, there wasn't really any obvious places where I felt that `CPython` was semantically better than `not(any(PyPy, GraalPy))`. Those implementations may align with CPython in the future, and is it then better to have e.g. `any(CPython, GraalPy)` or `not(PyPy)`?

By not having `cfg(CPython)` we force a consistent approach of just marking when code is diverging from CPython's path, and that seems workable for now, so I'll backtrack on this idea for the moment.